perf(linter): cache file extension

camchenry opened the pull request:
Getting file extensions shows up as slow in multiple rules. Computing it once and then storing it should be much faster.

<img width="713" height="366" alt="Screenshot 2025-10-20 at 12 00 35 AM" src="https://github.com/user-attachments/assets/b074bbf1-8a1f-4910-98d0-40f66eb3b39c" />

Boshen commented:
Interesting to see this is hot 🤯

camc314 commented:
### Merge activity

* **Oct 20, 8:09 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Oct 20, 8:09 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

camchenry committed:
perf(linter): cache file extension (#14794)

Getting file extensions shows up as slow in multiple rules. Computing it once and then storing it should be much faster.

<img width="713" height="366" alt="Screenshot 2025-10-20 at 12 00 35 AM" src="https://github.com/user-attachments/assets/b074bbf1-8a1f-4910-98d0-40f66eb3b39c" />
